src/cai_lab_to_nwb/zaki_2024/zaki_2024_convert_week_session.py

```diff
@@ -0,0 +1,98 @@
+"""Primary script to run to convert an entire session for of data using the NWBConverter."""
+
+import time
+from natsort import natsorted
+from pathlib import Path
+from typing import Union
+from datetime import datetime, timedelta
+import pandas as pd
+import json
+from neuroconv.utils import load_dict_from_file, dict_deep_update
+
+from zaki_2024_nwbconverter import Zaki2024NWBConverter
+
+
+def session_to_nwb(
+    data_dir_path: Union[str, Path],
+    output_dir_path: Union[str, Path],
+    subject_id: str,
+    stub_test: bool = False,
+    verbose: bool = True,
+):
+    print(f"Converting week-long session")
+    if verbose:
+        start = time.time()
+
+    data_dir_path = Path(data_dir_path)
+    output_dir_path = Path(output_dir_path)
+    if stub_test:
+        output_dir_path = output_dir_path / "nwb_stub"
+    output_dir_path.mkdir(parents=True, exist_ok=True)
+
+    nwbfile_path = output_dir_path / f"{subject_id}_week_session.nwb"
+
+    source_data = dict()
+    conversion_options = dict()
+
+    # Add EEG, EMG, Temperature and Activity signals
+    edf_folder_path = data_dir_path / "Ca_EEG_EDF" / (subject_id + "_EDF")
+    edf_file_paths = natsorted(edf_folder_path.glob("*.edf"))
+    if len(edf_file_paths) > 0:
```

Comment:
It seems to me that if the files are not found this should error immediately? after all, getting that data is the purpose of this interface.

Reply:
you're right! Thanks